### PR TITLE
storage: add support for generation on getSignedUrl

### PR DESCRIPTION
**Pull request by simonfan**

### CODE REVIEW

**Comment by simonfan** [comment]
I signed it!

File: lib/storage/file.js [hunk 1]
```diff
@@ -1173,6 +1173,8 @@ File.prototype.getSignedUrl = function(options, callback) {
   }[options.action];
 
   var name = encodeURIComponent(this.name);
+  var targetGeneration = is.undefined(this.generation) ?
+    false : encodeURIComponent(this.generation);
```
Comment (callmehiphop) [inline lib/storage/file.js]: Sorry for the nitpick, but could we move some of this logic around line 1226? I think we should wait to encode the generation until after `getCredentials` succeeds.

**Comment by callmehiphop** [comment]
Thanks for the pull request, @simonfan! I had a small nitpick on the code and it also appears that the lint task is failing (you can lint the code by running `npm run lint`), if you could address these items I would be happy to merge this change in! :)

File: lib/storage/file.js (hunk 1 above)
Comment (callmehiphop) [inline lib/storage/file.js]: Also, I think `generation` only comes in the form of integers, `encodeURIComponent` might not be necessary here.

**Comment by simonfan** [comment]
@callmehiphop great! I'll have a look on it later today and I'll let you know ;)

**Comment by simonfan** [comment]
There seems to be some issue with the [`.travis.yml before install`](https://github.com/GoogleCloudPlatform/gcloud-node/blob/8b6c4508f1c99278a9d6680831e86f4808b0a707/.travis.yml#L14-L16)  hook

The error at the travis build was:
```
0.00s$ export CXX=g++-4.8
0.00s$ $CXX --version
/home/travis/build.sh: line 45: g++-4.8: command not found
The command "$CXX --version" failed and exited with 127 during .
```

**Comment by stephenplusplus** [comment]
Hey @simonfan, thanks for this PR! I restarted the test, and that error seems to have disappeared. Travis looks happy now, so back to you, @callmehiphop!

**Comment by callmehiphop** [comment]
@simonfan looks great! Would you mind squashing your commits?

**Comment by callmehiphop** [comment]
Thanks @simonfan! 

:dancer:

**Comment by simonfan** [comment]
;)

**Pull request merged**